### PR TITLE
Change Request to Resume interrupted download of over-the-air

### PR DESCRIPTION
**Pull request by madhawap**
If the device reboots while downloading an OTA image, with this fix it will resume the firmware upgrade functionality.

### CODE REVIEW

File: system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java [hunk 1]
```diff
@@ -365,59 +365,70 @@ private void deleteRecursive(File fileOrDirectory, int level) throws IOException
...
-                // Set the title of this download, to be displayed in notifications
-                if (Constants.OTA_DOWNLOAD_PROGRESS_BAR_ENABLED) {
-                    request.setVisibleInDownloadsUi(true);
-                    request.setTitle("Downloading firmware upgrade");
-                    request.setDescription("WSO2 Agent");
-                    request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE);
+                    downloadReference = downloadManager.enqueue(request);
+                    Preference.putLong(context, context.getResources().getString(R.string.download_Manager_reference_id), downloadReference);
+                    Preference.putBoolean(context, context.getResources().getString(R.string.download_Manager_reference_id_available), true);
+                    Log.e(TAG, String.valueOf(Preference.getLong(context, context.getResources().getString(R.string.download_Manager_reference_id))));
```
Comment (charithag) [inline system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java]: Remove unnecessary error log

File: system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java [hunk 2]
```diff
@@ -516,6 +528,7 @@ public void run() {
                                 downloading = false;
                                 downloadManager.remove(downloadReference);
                                 cursor.close();
+                                Preference.putBoolean(context, context.getResources().getString(R.string.download_Manager_reference_id_available), false);
```
Comment (charithag) [inline system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java]: Remove redundant preference put

File: system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java (hunk 1 above)
Comment (madhawap) [reply system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java]: Fixed.

File: system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java (hunk 2 above)
Comment (madhawap) [reply system-service/app/src/main/java/org/wso2/iot/system/service/api/OTAServerManager.java]: Fixed.

**Pull request merged**